models/llm_openai/model.py

```
@@ -0,0 +1,76 @@
+from models.base import BaseModel
+from schemas.responses import MoralOutrage
+from lib.timestamp_utils import get_current_timestamp
+
+load_dotenv(override=True)
+
+class OpenAIModel(BaseModel):
+
+    def __init__(self, api_key: str | None = None) -> None:
+        self.api_key = api_key or os.getenv("OPENAI_API_KEY")
+        if not self.api_key:
+            raise ValueError("OpenAI API Key not found. Pass it to the constructor or set OPENAI_API_KEY environment variable.")
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)